src/event.rs

```diff
@@ -204,6 +204,13 @@ pub enum WindowEvent {
     ///
     /// For more information about DPI in general, see the [`dpi`](../dpi/index.html) module.
     HiDpiFactorChanged(f64),
+
+    /// The window theme has changed into (or out of) Dark Mode.
+    ///
+    /// The parameter is true if the window is now in a dark theme, and false if it is not.
+    /// Applications might wish to react to this to change the theme of the content of the window
+    /// when the system changes the window theme.
+    DarkModeChanged(bool),
```

Comment:
I don't think we should have this be a `bool`, since I can see a there isn't a clear semantic mapping between `true`/`false` and `Dark`/`Light`. Could we have this be `ThemeChanged(Theme)` where `Theme` is defined as follows?

```rust
enum Theme {
    Light,
    Dark,
}
```